service/javadsl/api/src/main/java/com/lightbend/lagom/javadsl/api/Descriptor.java

```diff
@@ -390,14 +390,14 @@ public String toString() {
     /**
      * Describes a topic call.
      */
-    public static final class TopicCall<Message> {
+    public static final class TopicCall<MESSAGE> {
```

Comment:
Reading the Java guidelines on this, all caps is not the naming convention:

> By convention, type parameter names are single, uppercase letters. This stands in sharp contrast to the variable naming conventions that you already know about, and with good reason: Without this convention, it would be difficult to tell the difference between a type variable and an ordinary class or interface name.

A single capital letter is the naming convention.  But in my opinion, that's a terrible naming convention, what would we do for ServiceCall for the request message and response message?  Both start with R.  It would be incredibly confusing reading code that just used a single letter there, for exactly the same reason that we don't use single letter variable names.

I don't understand the point about confusing with concrete class names - all Java developers use IDEs, and all IDEs highlight generic parameters in some way.  And at the end of the day, does it matter if it's a generic parameter or not when you're reading the code?  It's the name of a type, whether that's generic or not doesn't change much at all about the code that uses it, and as with everything else, Ctrl+click on it and you'll be taken straight to its definition if you want to learn more about it.